### PR TITLE
✨ feat(badges): add positioned badge support to badges

### LINKED ISSUE #44: ✨ Positioned badges support
**Is your feature request related to a problem? Please describe.**
[I saw in the docs](https://sveltestrap.js.org/?path=/docs/components-badges--docs) that sveltstrap supports Badges, but I couldn't find [Positioned Badges](https://getbootstrap.com/docs/5.3/components/badge/#positioned). They were added in Bootstrap v5. Is there support for it?

**Describe the solution you'd like**
I'd like to know if there's support for it or plans to support it. If they are supported, they are not in the docs.

**Describe alternatives you've considered**
I tried to use it with the example below, and it seems supported. So, it could be missing in the docs. Although this is not strictly a component implementation, it seems to follow Bootstrap's docs.
```svelte
      <Container>
          <Row>
            <Col>
              <p class="position-relative">
                <b>Title</b>
                <span class="position-absolute top-0 start-100 translate-middle badge rounded-pill bg-danger">
                  99+
                  <span class="visually-hidden">unread messages</span>
                </span>
              </p>
            </Col>
          </Row>
      </Container>
```

**Additional context**
Positioned badges examples
image
image

Thank you!

### PR DESCRIPTION
**Pull request by dysfunc**
**Why these changes?**
These changes add support for Bootstrap positioned badges via the `<Badge>` component. The `badge` interface has new properties:

- `ariaLabel`: The text will be read by screen readers
- `border`: If set, will add a light/dark border to the badge
- `indicator`: If set, it will assign additional padding and a border to the circular marker. Needs `positioned` to be `true`
- `placement`: The CSS bootstrap classes to determine placement. This defaults to `top-0 start-100`
- `positioned`: If set, it will assign the absolute positioning classes to the `badge`

```svelte
<script lang="ts">
  import { Button, Badge } from '@sveltestrap/sveltestrap';
</script>

<Button color="primary" class="position-relative">
  Inbox <Badge color="danger" pill positioned ariaLabel="Unread messages">100+</Badge>
</Button>

<Button color="primary" class="position-relative">
  Profile <Badge color="danger" pill border indicator positioned ariaLabel="New alerts"></Badge>
</Button>
```
Closes #44 

**Screenshots**
Screenshot 2024-01-04 at 9 17 27 AM

### CODE REVIEW

File: src/Badge/Badge.svelte [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
 
+  /**
+   * Text to be read by screen readers.
+   * @type {string}
+   */
+  export let ariaLabel = '';
+
+  /**
+   * Determines if the badge should have a border
+   * @type {string}
```
Comment (planetsLightningArrester) [inline src/Badge/Badge.svelte:12]: Shouldn't this be `@type {boolean}`?

File: src/Badge/Badge.svelte [hunk 2]
```diff
@@ -26,13 +38,40 @@
    */
   export let href = '';
 
+  /**
+   * Create a circular indicator for absolute positioned badge.
+   * @type {string}
```
Comment (planetsLightningArrester) [inline src/Badge/Badge.svelte:43]: Shouldn't this be `@type {boolean}` too?

File: src/Badge/Badge.svelte (hunk 1 above)
Comment (dysfunc) [reply src/Badge/Badge.svelte:12]: Yes, it should. Copy pasta 🍝

File: src/Badge/Badge.svelte (hunk 2 above)
Comment (dysfunc) [reply src/Badge/Badge.svelte:43]: Yes, it should. Copy pasta 🍝

File: src/Badge/Badge.d.ts [hunk 3]
```diff
@@ -3,9 +3,14 @@ import { HTMLAnchorAttributes } from 'svelte/elements';
 import { Color } from '../shared';
 
 export interface BadgeProps extends HTMLAnchorAttributes {
+  ariaLabel?: string;
+  border?: boolean;
```
Comment (bestguy) [inline src/Badge/Badge.d.ts:7]: Minor, but this seems a slippery slope as some users may rather use shadow, etc.   The styling may be better for people to customize in their custom theme (e.g. our company theme used all UPPERCASE in badges, defined in css build).

Would this be best to define in user themes or pass via existing `class` prop?

**Review by bestguy — COMMENTED** [review]
One question, otherwise looks good.

File: src/Badge/Badge.d.ts (hunk 3 above)
Comment (dysfunc) [reply src/Badge/Badge.d.ts]: Users will have the ability to do both. This was meant to provide support for the most common and basic usage. I've  actually added `shadow` as an option and modified both `shadow` and `border` to accept a `boolean` or any valid class for each prop. This are optional and further customization can be done via assigning class names to the `class` prop, adjusting the theme, or not including them.

**Pull request merged**